docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,155 @@
+
+## Performance considerations
+
+Tailspin created the ASP.NET Core services for *Tailspin.Web* and *Tailspin.Web.Surveys.Public* using Visual Studio templates. By default, these templates include logging to the console. Logging to the console may be done during development and debugging, but all logging to the console should be removed when the application is deployed to production.
+
+> [!NOTE]
+> For more information about setting up monitoring and diagnostics for Service Fabric applications running in production, see [monitoring and diagnostics][monitoring-diagnostics] for Azure Service Fabric.
+
+For example, the following lines in *startup.cs* for each of the web front end services should be commented out:
+
+```csharp
+// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+{
+    //loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+    //loggerFactory.AddDebug();
+
+    app.UseMvc();
+}
+```
+
+> [!NOTE]
+> These lines may be conditionally excluded when Visual Studio is set to “release” when publishing.
+
+Finally, when Tailspin deploys the Tailspin application to production, they switch Visual Studio to **release** mode.
+
+## Deployment considerations
+
+The refactored Surveys application is composed of five stateless services and one stateful service, so cluster planning is limited to determining the correct VM size and number of nodes. In the *applicationmanifest.xml* file that describes the cluster, Tailspin sets the *InstanceCount* attribute of the *StatelessService* tag to -1 for each of the services. A value of -1 directs Service Fabric to create an instance of the service on each node in the cluster.
+
+> [!NOTE]
+> Stateful services require the additional step of planning the correct number of partitions and replicas for their data.
+
+Tailspin deploys the cluster using the Azure Portal. The Service Fabric Cluster resource type deploys all of the necessary infrastructure, including VM scale sets and a load balancer. The recommended VM sizes are displayed in the Azure portal during the provisioning process for the Service Fabric cluster. Note that because the VMs are deployed in a VM scale set, they can be both scaled up and out as user load increases.
+
+> [!NOTE]
+> As discussed earlier, in the migrated version of the Surveys application the two web front ends were self-hosted using ASP.Net Core and Kestrel as a web server. While the migrated version of the Survey application does not use a reverse proxy, it is strongly recommended to use a reverse proxy such as IIS, Nginx, or Apache. In the refactored Surveys application, the two web front ends are self-hosted using ASP.Net Core with WebListener as a web server so a reverse proxy is not necessary.
```

Comment:
The reasoning here isn't completely clear if you don't know the history of Kestrel and WebListener. 

Also, the ASP.NET docs say "The 1.x versions of Kestrel don't have a full complement of defenses against attacks. This includes but isn't limited to appropriate timeouts, size limits, and concurrent connection limits." - and this statement is omitted from the Kestrel 2.0 docs. (Which I believe is intentional, see [this commit](https://github.com/aspnet/Docs/commit/424127e1ba0cc97cd6322c0613b6231c97c1b399#diff-165a51cd22ac4774fd788da0d54a6d39))

But the WebListener docs say "Because it's built on Http.Sys, WebListener doesn't require a reverse proxy server for protection against attacks."

So maybe our doc is fine as written ... maybe just point to the ASP.NET docs?

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?tabs=aspnetcore1x

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/weblistener

Reply:
Also - I don't know if this is worth calling out at all, but WebListener was renamed to HTTP.sys in 2.0 

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/httpsys